.github/scripts/trymerge.py

```diff
@@ -1109,6 +1109,35 @@ def __init__(self, message: str, rule: Optional['MergeRule'] = None) -> None:
         super().__init__(message)
         self.rule = rule
 
+class MergeError(Exception):
+    def __init__(self, reason: Union[str, Exception], internal_info: Optional[str] = None, title: str = "Merge failed"):
+        exception = f"**Reason**: {reason}"
+
+        run_url = os.getenv("GH_RUN_URL")
+        internal_debugging = ""
+        if internal_info is not None:
+            internal_debugging += internal_info + "\n\n"
+        if run_url is not None:
+            internal_debugging += f"Raised by <a href=\"{run_url}\">workflow job</a>"
+
+        if len(internal_debugging) > 0:
+            # Hide this behind a collapsed bullet since it's not helpful to most devs
+            internal_debugging_formatted = "\n".join((
+                "<details><summary>Details for Dev Infra team</summary>",
+                "",
+                f"{internal_debugging}"
+                "",
+                "</details>"
+            ))
+
+        msg = "\n".join((
+            f"## {title}",
+            f"{exception}",
+            "",
+            f"{internal_debugging_formatted}"
```

Comment:
`internal_debugging_formatted` would be None if `len(internal_debugging) == 0` wouldn't it?